Improve `RegexBasedFilter` doc (#1463)
jenkinsci/gitlab-plugin: README.md

## Advanced features
### Branch filtering
Triggers may be filtered based on the branch name, i.e. the build will only be allowed for selected branches. On the project configuration page, when you configure the GitLab trigger, you can choose 'Filter branches by name' or 'Filter branches by regex.' Filter by name takes comma-separated lists of branch names to include and/or exclude from triggering a build. Filter by regex takes a Java regular expression to include and/or exclude. For example, to exclude all branches containing the word "feature", you can use the following regular expression: `^(?:(?!feature).)*$`.
On a similar note, the regular expression `^(?!.*master).*$` will mean - all branches not matching master. This is a regular expression that uses negative lookahead to match any string that does not contain the word "master". Here's a breakdown of how it works:

    ^: Anchors the match to the beginning of the string.
    (: Starts a group that will be used for the negative lookahead.

>     ?!: Indicates a negative lookahead assertion - find all that doesnt match.

Reviewer: ```suggestion
    ?!: Indicates a negative lookahead assertion - find all that does not match.
```

    .*: Matches any number of characters (except for newline) zero or more times.
    master: should not match master.
    ): Ends the group.
    $: Anchors the match to the end of the string.
    
Keep in mind that the RegexBasedFilter feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
    

> Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the gitlab trigger:

Reviewer: ```suggestion
Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the GitLab trigger:
```

> ```triggers {

Reviewer: ````suggestion
```
triggers {
````

Reviewer: Formatting is a bit off here

## Advanced features
### Branch filtering
Triggers may be filtered based on the branch name, i.e. the build will only be allowed for selected branches. On the project configuration page, when you configure the GitLab trigger, you can choose 'Filter branches by name' or 'Filter branches by regex.' Filter by name takes comma-separated lists of branch names to include and/or exclude from triggering a build. Filter by regex takes a Java regular expression to include and/or exclude. For example, to exclude all branches containing the word "feature", you can use the following regular expression: `^(?:(?!feature).)*$`.
On a similar note, the regular expression `^(?!.*master).*$` will mean - all branches not matching master. This is a regular expression that uses negative lookahead to match any string that does not contain the word "master". Here's a breakdown of how it works:

    ^: Anchors the match to the beginning of the string.
    (: Starts a group that will be used for the negative lookahead.

>     ?!: Indicates a negative lookahead assertion - find all that does not match.

Reviewer: ```suggestion
    ?!: Indicates a negative lookahead assertion - finds all that does not match.
```

>     .*: Matches any number of characters (except for newline) zero or more times.

Reviewer: ```suggestion
    .*: Matches any number of characters (except for a newline) zero or more times.
```

    master: should not match master.
    ): Ends the group.
    $: Anchors the match to the end of the string.
    

> Keep in mind that the RegexBasedFilter feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.

Reviewer: ```suggestion
Keep in mind that the `RegexBasedFilter` feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
```

> Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the GitLab trigger:

Reviewer: ```suggestion
Here is an example pipeline script that shows how to use the `RegexBasedFilter` feature in the GitLab trigger:
```
